lib/overcommit/hook/prepare_commit_msg/replace_branch.rb

```diff
@@ -0,0 +1,54 @@
+module Overcommit::Hook::PrepareCommitMsg
+  # Prepends the commit message with a message based on the branch name.
+  # It's possible to reference parts of the branch name through the captures in
+  # the `branch_pattern` regex.
+  class ReplaceBranch < Base
+    def run
+      return :pass unless commit_msg_source == :commit
+      Overcommit::Utils.log.debug(
+        "Checking if '#{Overcommit::GitRepo.current_branch}' matches #{branch_pattern}"
+      )
+      if branch_pattern.match(Overcommit::GitRepo.current_branch)
+        prepend_commit_message
+        :pass
+      else
+        :warn
+      end
+    end
+
+    def prepend_commit_message
+      Overcommit::Utils.log.debug("Writing #{commit_msg_filename} with #{new_template}")
+      old_contents = File.read(commit_msg_filename)
+      File.open(commit_msg_filename, 'w') do |commit_file|
```

Comment:
Perhaps this code should be extracted into a utility function in `base`? `prepend_commit_msg` seems like a common-enough occurrence.